pkg/clients/helm/client.go

```diff
@@ -221,9 +221,11 @@ func (hc *client) pullChart(spec *v1beta1.ChartSpec, creds *RepoCreds, chartDir
 
 	pc.DestDir = chartDir
 
-	err := hc.login(spec, creds, pc.InsecureSkipTLSverify)
-	if err != nil {
-		return err
+	if pc.Username != "" || pc.Password != "" {
```

Comment:
should this be an `&&` check instead? i.e. I would expect both the user name and the password to need to exist in order for a login operation to succeed. What do you think?

Reply:
> I would expect both the user name and the password to need to exist in order for a login operation to succeed.

Probably 😄. I changed the code accordingly.